Cinema system/Cinema system/Startup.cs

```diff
@@ -29,9 +32,11 @@ public void ConfigureServices(IServiceCollection services)
                 configuration.RootPath = "ClientApp/dist";
             });
 
-             services.AddDbContext<CinemaDBContext>(options => {
+            services.AddScoped<ICinemaService, CinemaService>();
```

Comment:
So, in the cinema service I use DbContext and it intended to be consumed by controllers.  If i have a DbContext with a singleton lifetime, then all records that you read once will be tracked by the DbContext. This will require much more memory usage and it will keep growing.
With scoped, the DbContext has a transaction scope that's as long as the request.